File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.cpp
#include <bits/stdc++.h>
using namespace std;

vector<int> solution(vector<int> arr) 
{
    vector<int> answer;
    int before = arr[0];
    for(int i = 1; i < arr.size(); i++){
        if(arr[i] != before) {
            answer.push_back(before);
            before = arr[i];
        }
    }
    answer.push_back(arr.back());

    return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.*;

public class Solution {
    public int[] solution(int []arr) {
        ArrayList<Integer> list = new ArrayList<>();
        int before = arr[0];
        list.add(before);
        for(int num : arr){
            if(num == before) continue;
            before = num;
            list.add(before);
        }
        return list.stream().mapToInt(i->i).toArray();
    }
}